### PR TITLE
Add covalent radii and 3d drawing utilities

### LINKED ISSUE #46: Any interest in 3d display?
Using Makie it's possible to create 3d plots:

```julia
julia> calotte(mol; H=false,markersize=2)
```
image

(This would be improved by having a table of atomic radii.)

```julia
julia> ballstick(mol)
```
image

If you're interested I could make a pull request. There are really three options wrt implementation:
- we could make Makie a hard dependency. That would likely have negative consequences for both load time and installability due to the requirement for OpenGL.
- we could put it inside a `@require Makie` block using [Requires](https://github.com/JuliaPackaging/Requires.jl)
- we could make it a separate package that depends on both MolecularGraph and Makie

I suspect the latter two are the best options. If you're interested, which do you prefer? One concern about the Requires approach is that it's not currently compatible with precompilation, but putting it in a separate package would allow that. Makie itself is not very well precompilable now, so at present it wouldn't really make any difference, but hopefully Makie will get more precompilable over time.

CC @SimonDanisch

### PR DESCRIPTION
**Pull request by timholy**
Closes #46

### CODE REVIEW

File: src/model/atom.jl [hunk 1]
```diff
@@ -8,14 +8,43 @@ export
     setcharge, setstereo, atomnumber, todict
 
 
-const ATOMTABLE = YAML.load(open(
-    joinpath(dirname(@__FILE__), "../../assets/const/atomicweights.yaml")
-))
+const ATOMTABLE = let
+    weightsfile = joinpath(dirname(@__FILE__), "../../assets/const/atomicweights.yaml")
+    include_dependency(weightsfile)
```
Comment (timholy) [inline src/model/atom.jl:13]: Using `include_dependency` will cause the package to be recompiled if the data change. Seemed like a useful addition to me.

File: src/draw/draw3d.jl [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+using .AbstractPlotting
+
+export spacefilling, ballstick
+
+atomcolors = Dict("O" => colorant"red",
+                  "C" => colorant"black",
+                  "H" => colorant"lightgray",
+                  "S" => colorant"yellow",
+                  "P" => colorant"green")
```
Comment (timholy) [inline src/draw/draw3d.jl:16]: Happy to add more colors to this list. Just let me know what you want.

Comment (mojaie) [reply src/draw/draw3d.jl:16]: How about `"N" => colorant"blue",`?

Here is the colors in 2D depiction
https://github.com/mojaie/MolecularGraph.jl/blob/aef2b1d4f02e508256d85e54392d97b5b45a97fa/src/draw/draw2d.jl#L35-L49

File: src/model/atom.jl (hunk 1 above)
Comment (mojaie) [reply src/model/atom.jl:13]: I did not know that, it looks very useful. Thank you!

**Comment by timholy** [comment]
Worth noting that in addition to implementing the 3d rendering, this reworks Project.toml dependencies, splitting out any test dependencies into `test/Project.toml`.

**Comment by timholy** [comment]
Another question is whether to add an image of the 3d plotting to the README.

File: src/draw/draw3d.jl (hunk 2 above)
Comment (timholy) [reply src/draw/draw3d.jl]: I incorporated these. One oddity is `:H` being black: that seems different from CPK coloring, https://en.wikipedia.org/wiki/CPK_coloring.

**Review by mojaie — APPROVED** [review]
Looks great. Thank you very much!

File: src/draw/draw3d.jl (hunk 2 above)
Comment (mojaie) [reply src/draw/draw3d.jl]: Yes,  :H has been black for visibility on white background, but gray would be a good alternative. I want to consider making atom color set options from several CPK colors.

**Comment by timholy** [comment]
Thanks for the approval!

I added a few more things, particularly van der Waals radii for all the main-group elements. Because of the name `spacefilling`, I now default to using the van der Waals radius but also support the former covalent radii. I also display hydrogen in white and support overlays of multiple molecules.

I don't have the ability to merge this PR, so if you agree with these changes then I'll need you to do the merge.

**Review by mojaie — APPROVED** [review]

**Comment by mojaie** [comment]
Thank you very much!

**Pull request merged**